epitran/data/post/kor-Hang.txt

```diff
@@ -1,42 +1,84 @@
 ::initialconsonants:: = k|kʰ|k͈|n|t|t͈|ɾ|m|p|p͈|s|s͈|t͡ɕ|t͡ɕ͈|t͡ɕʰ|tʰ|pʰ|h
-::vowels:: = i|e|ø|we|ɛ|a|o|u|ʌ|ɯ|ɰi|je|jɛ|ja|jo|ju|jʌ|ɥi|wi|we|wɛ|wa|wʌ
-% (not used) ::finalconsonants:: = k̚|ŋ|t̚|n|l|p̚|m
+::vowels:: = i|e|ɛ|a|o|u|ʌ|ɯ|ɰi|je|jɛ|ja|jo|ju|jʌ|wi|we|wɛ|wa|wʌ
+::nasals:: = n|m
+::finalObs:: = k|t|p
+::finalforNasal:: = k|t|m|p|ŋ
+::jvowel:: = i|je|jɛ|ja|jo|ju|jʌ
 
-% remove ㅇ if followed by a vowel
+% remove ㅇ if followed by a vowel (if it is in initial consonant position)
 ŋ -> 0 / _ (::vowels::)
 
-% change to medial allophone if surrounded by vowels
-k -> g / (::vowels::) _ (::vowels::)
-t -> d / (::vowels::) _ (::vowels::)
-t͡ɕ -> dʑ / (::vowels::) _ (::vowels::)
-p -> b / (::vowels::) _ (::vowels::)
-l -> ɾ / (::vowels::) _ (::vowels::)
+% change ɰi to i if preceded by consonant
+ɰi -> i / (::initialconsonants::) _
 
+% Substitution of lax to aspirated for consonant clusters with 'h'
+nht -> ntʰ / (::vowels::) _ (::vowels::)
+nhk -> nkʰ / (::vowels::) _ (::vowels::)
+nht͡ɕ -> nt͡ɕʰ / (::vowels::) _ (::vowels::)
+lht -> lhtʰ / (::vowels::) _ (::vowels::)
+lhk -> lhkʰ / (::vowels::) _ (::vowels::)
+lht͡ɕ -> lht͡ɕʰ / (::vowels::) _ (::vowels::)
+lph -> lpʰ / (::vowels::) _ (::vowels::)
+lkh -> lkʰ / (::vowels::) _ (::vowels::)
+nt͡ɕh -> nt͡ɕʰ / (::vowels::) _ (::vowels::)
 
-% change to final allophone if followed by an initialconsonant
-k̚s͈ -> k̚ / _ (::initialconsonants::|#)
-lɡ -> k̚ / _ (::initialconsonants::|#)
-ndʑ -> n / _ (::initialconsonants::|#)
+% Substitution of lax to tensce for consonant clusters with 's'
+ks -> ks͈ / (::vowels::) _ (::vowels::)
+ls -> ls͈ / (::vowels::) _ (::vowels::)
+ps-> ps͈ / (::vowels::) _ (::vowels::)
+
+% Simplification of consonant clusters to single consonants at coda 
```

Comment:
Shouldn't these also apply word-finally?

Reply:
The second consonant in a coda cluster shifts to the next syllable if followed by a vowel, except /s/, which becomes /s͈/. I'll update the annotation descriptions to clarify this rule.